### PR TITLE
Add pull to refresh to list views

### LINKED ISSUE #365: Add pull to refresh to table and collection views
**Affected Component:**
- DashboardViewController
- CourseListViewController
- AnnouncementsViewController
- PlatformEventsViewController

**Expected Behavior:**
- refresh views via pull to refresh elements

### PR DESCRIPTION
**Pull request by mathebox**
Fixes #365

Add pull to refresh to the following views:
- CourseDatesVC (incl. CourseActivityVC)
- CourseListVC
- CourseContentVC
- AnnouncmentVC
- PlatformEventVC

### CODE REVIEW

File: xikolo-ios/ViewControllers/Dashboard/CourseDatesTableViewController.swift [hunk 1]
```diff
@@ -63,10 +72,25 @@ class CourseDatesTableViewController : UITableViewController {
 
     func updateAfterLoginStateChange() {
         self.navigationItem.rightBarButtonItem = UserProfileHelper.isLoggedIn() ? nil : self.loginButton
+        self.refresh()
+    }
 
+    func refresh() {
         self.tableView.reloadEmptyDataSet()
+        let deadline = 750.milliseconds.fromNow
+        let endSpinning = {
```
Comment (bjrne) [inline xikolo-ios/ViewControllers/Dashboard/CourseDatesTableViewController.swift]: Is there a reason for calling this one `endSpinning` when both other occurrences of the same code are `stopRefreshControl`?

File: xikolo-ios/ViewControllers/Dashboard/CourseDatesTableViewController.swift [hunk 2]
```diff
@@ -63,10 +72,25 @@ class CourseDatesTableViewController : UITableViewController {
 
     func updateAfterLoginStateChange() {
         self.navigationItem.rightBarButtonItem = UserProfileHelper.isLoggedIn() ? nil : self.loginButton
+        self.refresh()
+    }
 
+    func refresh() {
         self.tableView.reloadEmptyDataSet()
+        let deadline = 750.milliseconds.fromNow
```
Comment (bjrne) [inline xikolo-ios/ViewControllers/Dashboard/CourseDatesTableViewController.swift]: I guess this number is just randomly chosen?

Comment (mathebox) [reply xikolo-ios/ViewControllers/Dashboard/CourseDatesTableViewController.swift]: I found this time provides the a good UX (minimum time the refresh control is shown). Maybe this should be a constant.

File: xikolo-ios/ViewControllers/Dashboard/CourseDatesTableViewController.swift (hunk 1 above)
Comment (mathebox) [reply xikolo-ios/ViewControllers/Dashboard/CourseDatesTableViewController.swift]: Yeah, this should also be `stopRefreshControl `.

**Pull request merged**